### PR TITLE
Add jest, typedoc, cyclomatic complexity warnings

### PR DESCRIPTION
**Pull request by michielbdejong**
This PR supersedes #188 and #189. It's on Sharon's fork so she and I can both work on it.

### CODE REVIEW

**Comment by michielbdejong** [comment]
@SharonStrats did you mean to commit `src/playground.js` or was that an accident?
10:53
I had to comment it out to get the tests to pass

**Comment by michielbdejong** [comment]
Next, I'll mock rdflib and solid-auth-client so that we can write better unit tests for the actual solid-ui code, without triggering network requests.

**Comment by michielbdejong** [comment]
The build is failing because it doesn't have access to the API token,  guess maybe that's because this branch is on Sharon's account. https://travis-ci.com/SharonStrats/solid-ui/jobs/283995290

**Comment by michielbdejong** [comment]
In https://travis-ci.org/solid/solid-ui/builds/646637282 it passes

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (michielbdejong) [inline .vscode/launch.json]: Unfortunately you can't use comments in JSON files (yes, I know).

File: src/pad.js [hunk 2]
```diff
@@ -27,9 +27,9 @@ const utils = require('./utils')
  * @param {NamedNode} author - The author of text being displayed
  * @returns {String} The CSS color generated, constrained to be light for a background color
  */
-UI.pad.lightColorHash = function (author) {
-  var hash = function (x) {
-    return x.split('').reduce(function (a, b) {
+UI.pad.lightColorHash = function(author) {
+  var hash = function(x) {
+    return x.split('').reduce(function(a, b) {
```
Comment (michielbdejong) [inline src/pad.js]: Do we have a lint rule enabled for this spacing? If not, we should!

Comment (michielbdejong) [reply src/pad.js]: Ah wait, we do. Try running `npm run lint` on your code, you'll see it complain:
Screenshot 2020-02-10 09 01 29

File: .vscode/launch.json (hunk 1 above)
Comment (Vinnl) [reply .vscode/launch.json]: Luckily VSCode [doesn't stick to the JSON spec](https://code.visualstudio.com/Docs/languages/json#_json-with-comments) here :)

**Comment by michielbdejong** [comment]
I'll help @SharonStrats rebase here master branch on the current master and then create a new PR for this.

**Pull request closed without merging**